pg_search/src/postgres/utils.rs

```diff
@@ -164,16 +164,16 @@ pub unsafe fn row_to_search_document(
 
         if *is_array {
             for value in TantivyValue::try_from_datum_array(datum, *base_oid)? {
-                document.insert(search_field.id, value.tantivy_schema_value());
+                document.insert(search_field.id, value.into());
```

Comment:
This was making a clone of the backing OwnedValue, which turned out not to be necessary with a proper trait impl.

This could have been a contributor to some transient high-memory usage as it would have duplicated the backing OwnedData for some short period of time, which could have been a lot for tables with large strings or json objects or arrays.